src/Orleans/Runtime/CallbackData.cs

```diff
@@ -70,20 +93,18 @@ public void StartTimer(TimeSpan time)
             }
 
             TimeSpan firstPeriod = timeout;
-            TimeSpan repeatPeriod = Constants.INFINITE_TIMESPAN; // Single timeout period --> No repeat
             if (config.ResendOnTimeout && config.MaxResendCount > 0)
             {
-                firstPeriod = repeatPeriod = timeout.Divide(config.MaxResendCount + 1);
+                resendPeriod = firstPeriod = timeout.Divide(config.MaxResendCount + 1);
+                DueTime = DateTime.UtcNow.Add(firstPeriod);
```

Comment:
you can collapse this condition and the n on resend case into one, with single timer wheel var, and only difference via timerWheelShouldReEnqueue/shouldDequeue values. No need for code dup.